packages/itwinui-react/src/core/Tooltip/Tooltip.tsx

```diff
@@ -16,46 +71,148 @@ export type TooltipProps = {
+        animationFrame: updateOptions.animationFrame,
+        ancestorScroll: updateOptions.ancestorScroll,
+        ancestorResize: updateOptions.ancestorResize,
+        elementResize: updateOptions.elementResize,
+      }),
+    middleware: [
+      middleware.offset !== undefined ? offset(middleware.offset) : offset(4),
+      middleware.flip && flip(),
+      middleware.shift && shift(),
+      middleware.size && size(),
+      middleware.autoPlacement && autoPlacement(),
+      middleware.inline && inline(),
+      middleware.hide && hide(),
+    ].filter(Boolean),
+  });
+
+  const context = data.context;
+
+  const hover = useHover(context, {
+    enabled: controlledOpen == null,
+    delay: {
+      open: 50,
+      close: 250,
+    },
+    handleClose: safePolygon({ buffer: -Infinity }),
+  });
+
+  const focus = useFocus(context, {
+    enabled: controlledOpen == null,
+  });
+
+  const click = useClick(context, {
+    enabled: controlledOpen == null,
+  });
+
+  const dismiss = useDismiss(context, {
+    enabled: controlledOpen == null,
+  });
+
+  const role = useRole(context, { role: 'tooltip' });
+
+  const interactions = useInteractions([click, hover, focus, dismiss, role]);
+
+  return React.useMemo(
+    () => ({
+      open,
+      setUncontrolledOpen,
+      ...interactions,
+      ...data,
+    }),
+    [open, interactions, data],
+  );
+};
 
 /**
  * Basic tooltip component to display informative content when an element is hovered or focused.
- * Uses the {@link Popover} component, which is a wrapper around [tippy.js](https://atomiks.github.io/tippyjs).
+ * Uses [FloatingUI](https://floating-ui.com/).
  * @example
- * <Tooltip content='tooltip text' placement='top'><div>Hover here</div></Tooltip>
- * @example
- * const buttonRef = React.useRef();
- * ...
- * <Button ref={buttonRef} />
- * <Tooltip content='tooltip text' reference={buttonRef} />
+ * <Tooltip content='tooltip text' placement='top'>Hover here</Tooltip>
  */
-export const Tooltip = (props: TooltipProps) => {
-  const { content, children, className, style, visible, ref, id, ...rest } =
-    props;
+export const Tooltip = React.forwardRef((props, forwardRef) => {
+  const {
+    content,
+    children,
+    portal = true,
+    placement = 'top',
+    updateOptions,
+    middleware,
+    style,
+    className,
+    visible,
+    ...rest
+  } = props;
+  const tooltip = useTooltip({ placement, visible, updateOptions, middleware });
+  const context = useGlobals();
 
-  return (
-    <Popover
-      visible={visible}
-      interactive={false}
-      content={
-        <Box
-          className={cx('iui-tooltip', className)}
-          style={style}
-          role='tooltip'
-          id={id}
-        >
-          {content}
-        </Box>
-      }
-      offset={[0, 4]}
-      ref={ref}
+  const portalTo =
+    typeof portal !== 'boolean'
+      ? portal.to
+      : portal
+      ? context?.portalContainerRef?.current ?? getDocument()?.body
+      : null;
```

Comment:
This is a little hard for me to understand. Can you explain what this means and what is it for?

Reply:
prop `portal` can be boolean (true or false) or object with `to` value.

If portal is not boolean, we add it to the user defined element.
If portal is true - add it to our div that is set in theme provider or document body (if portalContainer is not there).
If portal is false - add it the tooltip next to the element.

Play around with these settings and see in dev tools where tooltip is added.